FusionIIIT/applications/academic_information/views.py

```diff
@@ -346,11 +347,12 @@ def add_curriculum(request):
                     optional=optional,
                     course_type=course_type,
                 )
+                new_curr.append(ins)
             else:
-                print("Khatam")
                 break
             i+=1
-        curriculum = Curriculum.objects.filter(branch = branch).filter(batch = batch).filter(programme= programme)
+        Curriculum.objects.bulk_create(new_curr)
+        curriculum = Curriculum.objects.select.related().filter(branch = branch).filter(batch = batch).filter(programme= programme)
```

Comment:
Still unfixed. Change `select.related()` to `select_related()`

Reply:
Done

Reply:
Include `user` and `department` fields in `select_related()` of optimized queries related to `ExtraInfo` model as `department` is a **nullable Foreign Key**. Do this in `academic_procecures/views.py` as well.